lib/http/apps/coin.js

```diff
@@ -31,4 +32,41 @@ app.post(`/mint`, (req, res) => {
     });
 });
 
+app.post(`/transfer`, (req, res) => {
+    let to = req.body.to;
+    let amount = req.body.amount;
+
+    res.locals.coinsence.Coin.functions.transfer(to, amount, { gasLimit: 5000000 }).then(transaction => {
+        console.log(transaction);
+        res.status(201).json(transaction);
+    }).catch(e => {
+        console.log(e);
+        res.sendStatus(500);
+    });
+});
+
+app.get(`/balance`, (req, res) => {
+    let owner = req.body.owner;
+
+    res.locals.coinsence.Coin.functions.balanceOf(owner).then((result) => {
+        let balanceHex = result._hex;
+        let balance = result.toString();
+        res.status(201).json({ balanceHex, balance });
+    }).catch(e => {
+        console.log(e);
+        res.sendStatus(500);
+    });
+});
+
+app.get(`/supply`, (req, res) => {
```

Comment:
maybe we should use the contract names. so in this case here also totalsupply?

Reply:
okay (y)